rt/frontend/jaxws/src/main/java/org/apache/cxf/jaxws/spi/WrapperClassUtils.java

```diff
@@ -0,0 +1,60 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.cxf.jaxws.spi;
+
+import java.lang.reflect.Method;
+import java.util.Locale;
+import java.util.regex.Pattern;
+
+public final class WrapperClassUtils {
+    public static final String DEFAULT_PACKAGE_NAME = "defaultnamespace";
+    private static final Pattern JAVA_PACKAGE_NAME_SANITIZER_PATTERN = Pattern.compile("[^a-zA-Z0-9_]");
+
+    private WrapperClassUtils() {
+    }
+
+    /**
+     * Returns a package name unique for the declaring class of the given {@code method} and {@code anonymous}
+     * parameters suitable for generated classes.
```

Comment:
@ppalaga thanks for javadoc, could you please document a few examples how package name is generated? For example (from tests):

```
org.apache.cxf.systest.jaxws.CXF9003Test$Hello1Service -> org.apache.cxf.systest.jaxws.jaxws_asm.cxf9003test_hello1service
```

Thank you.

Reply:
Addressed in 6582f43